src/gt4py/next/iterator/transforms/cse.py

```diff
@@ -365,10 +371,10 @@ class CommonSubexpressionElimination(PreserveLocationVisitor, NodeTranslator):
     Perform common subexpression elimination.
 
```

Comment:
Could you briefly outline here the algorithm and the used tools (e.g. other visitors, functions, ...)